### PR TITLE
[Bugfix] CallAdapter id checking

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Introduces blocking error when passing teams id to CallAdapter and acs id to Teams adapter
# Why

Stops bad id's from being added to the wrong adapters. this causes strange behaviour with CTE for example that we want to stop

https://skype.visualstudio.com/SPOOL/_workitems/edit/3541196
# How Tested

Tested locally

### CODE REVIEW

File: packages/react-components/src/components/Dialpad/Dialpad.test.tsx [hunk 1]
```diff
@@ -15,9 +15,35 @@ const onSendDtmfTone = (dtmfTone: DtmfTone): Promise<void> => {
   return Promise.resolve();
 };
 
-window.AudioContext = jest.fn().mockImplementation(() => {
-  return {};
-});
+const oscillatorNodeMocks = {
```
Comment (JamesBurnside) [inline packages/react-components/src/components/Dialpad/Dialpad.test.tsx:18]: remove unrelated code that got pulled in 😉

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts [hunk 2]
```diff
@@ -1452,6 +1451,11 @@ export const createAzureCommunicationCallAdapter = async ({
   /* @conditional-compile-remove(PSTN-calls) */ alternateCallerId,
   /* @conditional-compile-remove(video-background-effects) */ options
 }: AzureCommunicationCallAdapterArgs): Promise<CallAdapter> => {
+  if (isMicrosoftTeamsUserIdentifier(userId)) {
+    throw new Error(
+      'Microsoft Teams user identifier is not supported by AzureCommunicationCallAdapter, please use our TeamsCallAdapter.'
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:1456]: nitty nit: avoid possessive adjective
```suggestion
      'Microsoft Teams user identifier is not supported by AzureCommunicationCallAdapter. Instead use TeamsCallAdapter.'
```

**Review by JamesBurnside — APPROVED** [review]

**Review by prabhjot-msft — APPROVED** [review]
